src/coreclr/vm/eetwain.cpp

```diff
@@ -2144,9 +2144,8 @@ void EECodeManager::UpdateSSP(PREGDISPLAY pRD)
 #ifdef FEATURE_INTERPRETER
 DWORD_PTR InterpreterCodeManager::CallFunclet(OBJECTREF throwable, void* pHandler, REGDISPLAY *pRD, ExInfo *pExInfo, bool isFilter)
 {
-    // Interpreter-TODO: implement calling the funclet in the intepreted code
-    _ASSERTE(FALSE);
-    return 0;
+    InterpMethodContextFrame* faultingFrame = (InterpMethodContextFrame*)pRD->SP;
```

Comment:
I thought the plan was be to create a new frame only for filter funclets, because the stack wasn't yet unwound when they are called. For finally and catch funclets I thought we would be able to call them from the same exact frame without making use of this frame var.

Reply:
We have found it would not work with the coreclr. It would cause frames reported by the stack walk to not to be ordered by their stack location anymore. It would result in reporting the frame in whose context the finally was running, skipping several explicit frames like the InterpreterFrame and the pinvoke frame of the CallFinallyFunclet / CallCatchFunclet. Besides the ordering issue, skipping the CallFinallyFunclet / CallCatchFunclet would prevent the EH detecting that the exception has escaped from a funclet. The throw from finally was a case where we have found this was a problem.

Reply:
I am not 100% sure I understand what you meant. The first sequence seemed to indicate that we don't need a separate InterpMethodContextFrame for the funclets, but reading the rest seems the meaning was actually the opposite. If the point was that we can have a separate InterpMethodContextFrame for the funclets, but use the interpreter stack pointer of the parent frame of the funclet, then I agree. I've already mentioned that to Andrew before.

Reply:
I initially thought we might share also the `InterpMethodContextFrame`, but my original observation was more about sharing the method \'stack frame\' from the interpreter stack and avoiding the use of a separate base pointer.

>  If the point was that we can have a separate InterpMethodContextFrame for the funclets, but use the interpreter stack pointer of the parent frame of the funclet, then I agree. I've already mentioned that to Andrew before.

Sounds all good then